### PR TITLE
Update for breaking changes in IAP plugin

### PR DESCRIPTION
**Pull request by renefloor**
The in_app_purchases package will have some breaking changes soon. This PR adapts the codelab example for the coming breaking changes in flutter/plugins/pull/3877

### CODE REVIEW

File: in_app_purchases/complete/app/pubspec.lock [hunk 1]
```diff
@@ -187,17 +257,45 @@ packages:
   in_app_purchase:
     dependency: "direct main"
     description:
-      name: in_app_purchase
+      path: "../../../../flutter-plugins/packages/in_app_purchase/in_app_purchase"
```
Comment (domesticmouse) [inline in_app_purchases/complete/app/pubspec.lock:260]: I'm curious how this dependency path came about. Is the plug-in published yet?

File: in_app_purchases/complete/app/pubspec.yaml [hunk 2]
```diff
@@ -26,7 +26,8 @@ dependencies:
   cloud_functions: "^1.0.3"
   cloud_firestore: "^1.0.4"
   google_sign_in: "^5.0.1"
-  in_app_purchase: ^0.5.1+3
+  in_app_purchase:
+    path: ../../../../../flutter-core/flutter-plugins/packages/in_app_purchase/in_app_purchase
```
Comment (domesticmouse) [inline in_app_purchases/complete/app/pubspec.yaml:30]: This can't be published as is

File: in_app_purchases/step_06/app/pubspec.lock [hunk 3]
```diff
@@ -187,17 +257,45 @@ packages:
   in_app_purchase:
     dependency: "direct main"
     description:
-      name: in_app_purchase
+      path: "../../../../flutter-plugins/packages/in_app_purchase/in_app_purchase"
```
Comment (domesticmouse) [inline in_app_purchases/step_06/app/pubspec.lock:260]: Revert this too

File: in_app_purchases/step_06/app/pubspec.yaml [hunk 4]
```diff
@@ -26,7 +26,8 @@ dependencies:
   cloud_functions: "^1.0.3"
   cloud_firestore: "^1.0.4"
   google_sign_in: "^5.0.1"
-  in_app_purchase: ^0.5.1+3
+  in_app_purchase:
+    path: ../../../../../flutter-core/flutter-plugins/packages/in_app_purchase/in_app_purchase
```
Comment (domesticmouse) [inline in_app_purchases/step_06/app/pubspec.yaml:30]: Ditto

File: in_app_purchases/step_07/app/pubspec.yaml [hunk 5]
```diff
@@ -26,7 +26,8 @@ dependencies:
   cloud_functions: "^1.0.3"
   cloud_firestore: "^1.0.4"
   google_sign_in: "^5.0.1"
-  in_app_purchase: ^0.5.1+3
+  in_app_purchase:
+    path: ../../../../../flutter-core/flutter-plugins/packages/in_app_purchase/in_app_purchase
```
Comment (domesticmouse) [inline in_app_purchases/step_07/app/pubspec.yaml:30]: Ditto

File: in_app_purchases/step_08/app/pubspec.yaml [hunk 6]
```diff
@@ -26,7 +26,8 @@ dependencies:
   cloud_functions: "^1.0.3"
   cloud_firestore: "^1.0.4"
   google_sign_in: "^5.0.1"
-  in_app_purchase: ^0.5.1+3
+  in_app_purchase:
+    path: ../../../../../flutter-core/flutter-plugins/packages/in_app_purchase/in_app_purchase
```
Comment (domesticmouse) [inline in_app_purchases/step_08/app/pubspec.yaml:30]: Ditto

File: in_app_purchases/step_09/app/pubspec.yaml [hunk 7]
```diff
@@ -26,7 +26,8 @@ dependencies:
   cloud_functions: "^1.0.3"
   cloud_firestore: "^1.0.4"
   google_sign_in: "^5.0.1"
-  in_app_purchase: ^0.5.1+3
+  in_app_purchase:
+    path: ../../../../../flutter-core/flutter-plugins/packages/in_app_purchase/in_app_purchase
```
Comment (domesticmouse) [inline in_app_purchases/step_09/app/pubspec.yaml:30]: Ditto

**Review by domesticmouse — CHANGES_REQUESTED** [review]
I'm guessing you are waiting on an updated package to be published before we land this PR?

**Comment by renefloor** [comment]
> I'm guessing you are waiting on an updated package to be published before we land this PR?

Yep, the PR mentioned in the PR post needs to be released: flutter/plugins/pull/3877
The [platform interface](https://pub.dev/packages/in_app_purchase_platform_interface) is already published, the android, iOS and general implementations not yet.

**Comment by domesticmouse** [comment]
I'd suggest marking this PR as WIP until the packages are published.

File: in_app_purchases/complete/app/pubspec.yaml (hunk 2 above)
Comment (domesticmouse) [reply in_app_purchases/complete/app/pubspec.yaml:30]: Quick thought, if you replace this path dependency with a [git dependency](https://dart.dev/tools/pub/dependencies#git-packages), we should be able to get this PR past the CI pipeline. 

That would enable us to land this PR and then update the codelab. Thoughts?

**Review by domesticmouse — APPROVED** [review]

**Comment by kleeb** [comment]
any plans for making those samples working ?

```
src/iap.repository.ts:1:9 - error TS2724: '"firebase-admin/lib/firestore"' has no exported member named 'firestore'. Did you mean 'Firestore'?

import {firestore} from "firebase-admin/lib/firestore";
```

**Comment by domesticmouse** [comment]
> any plans for making those samples working ?
> 
> ```
> src/iap.repository.ts:1:9 - error TS2724: '"firebase-admin/lib/firestore"' has no exported member named 'firestore'. Did you mean 'Firestore'?
> 
> import {firestore} from "firebase-admin/lib/firestore";
> ```

I'm part way through rebuilding the codelab to use a Dart back end running on Cloud Run, which is more suitable for us to maintain going forward.

**Pull request merged**